### ISSUE TITLE
Electron 1.5 and above significant issues with Dragon voice operation (menus and performance)

### ISSUE THREAD

**Issue by cleidigh** [electron/electron#9215]
OS: Windows 7 professional
Electron app: Microsoft Visual Studio code (just upgraded electron to v1.6.2 from 1.4.6

- I am a contributor to Visual Studio code
- I am also disabled and utilize Dragon Naturally Speaking DPI 14 from Nuance (the de facto voice dictation program for Windows)
- After VSC upgraded to electron version 1.6.2 the ability to control submenus completely disappeared
- Main native menu items are only accessible if the window is maximized
- I am also seeing major performance degradation of Dragon controlling electron/VSC , but this is much harder to quantify and appears erratic

This is a major issue for me personally and I think detrimental to electron overall for accessibility
As a developer I would be very interested trying to analyze and help solving these issues.

To do some isolated experiments I utilized the electron hello app and built it with successive
electronic versions from 1.4.6 up to 1.6.2 as well as the mainline version
With the node and chrome changes mentioned in version 1.5 this is where the problems began.

Please let me know how I can help solve this problem.
Thanks Christopher

**Comment by kevinsawicki** [comment]
@cleidigh are you able to reproduce this issue in Chrome as well? Do the menus properly work there?

There are several open menu issues in Chrome reported as part of an a11y assessment, perhaps one of them is related to this, https://bugs.chromium.org/p/chromium/issues/list?can=1&q=A11y+Assessment

**Comment by cleidigh** [comment]
@kevinsawicki 
thanks for your response, sorry for my delayed working on a VSC Release.

I took a look at the list you sent and then did a search which pulled up the following:

https://bugs.chromium.org/p/chromium/issues/detail?id=98102

Yes Chrome does have some issues though they are slightly different but I'm sure related.

when you access the hamburger menu then try to use "Click <MenuItem>" the item above
the one desired is selected. if you try to click a bookmark bar item you can see the mouse cursor
move up and to the left. 

this behavior started I think somewhat less than a year ago. 

with a basic electron app with the traditional menu bar in Windows, it is possible to select
by voice the main menu items. However after selection and pop-up of the menu itself
Dragon slows to a crawl and cannot select any of the menu items yet it is possible
to see the Dragon can "see" the menu item names as a debug log shows detection
but I cannot select them. Unfortunately Nuance has been unreachable.

I was hoping there might be something at the electron menu level that would match what Dragon expects something I would be happy to try to help with. it seems since it half works
there might be something basic that can be done to rectify this.

I am a voice only disabled user and would really hate to see electron apps inaccessible
by users like myself. it's also cramping my work on VSC since I cannot access the menus normally
on the project work.

Thanks Christopher

**Comment by kevinsawicki** [comment]
> With the node and chrome changes mentioned in version 1.5 this is where the problems began.

So just to confirm, the 1.4.x line of Electron does not exhibit this behavior? 1.5.0 is the earlier version that it happened in?

**Comment by kevinsawicki** [comment]
>  if you try to click a bookmark bar item you can see the mouse cursor
move up and to the left.

If you create a bookmarks folder with a few bookmarks on it and then from the new tab page click the folder from the toolbar, can it be successfully navigated? I think this might be the closest menu example in Chrome that matches the Electron's menu bar usage.

**Comment by kevinsawicki** [comment]
> Main native menu items are only accessible if the window is maximized

Also, can you confirm if context menus are accessible? And it is only the menu bar that is affected.

**Comment by cleidigh** [comment]
@kevinsawicki 
- I did the hello app tests stepping up to 1.4.16 - I believe these were all consistent and worked correctly
 I can reconfirm these tests if necessary
- BeginningWithVersion 1.5 the offset issue started
the bookmark bar has the same issue, not sure when context menus broke

I'm trying to put together some videos as I think they will show things more clearly should be in the next day or so.

**Comment by zcbenz** [comment]
@cleidigh The menu bar in Electron uses the implementation of Chrome's bookmark bar, this seems to be a regression of Chrome. Do you mind reporting to Chromium's issues list on accessibility not working with bookmark bar?

**Comment by codebytere** [comment]
We are no longer implementing bugfixes for versions of Electron <= `1.7.x`, so i'm going to close this issue but if it is still persisting in more recent versions of Electron we can certainly reopen it!
